PyPDF2/_reader.py

```diff
@@ -326,8 +326,11 @@ def metadata(self) -> Optional[DocumentInformation]:
             return None
         obj = self.trailer[TK.INFO]
         retval = DocumentInformation()
-        retval.update(obj)  # type: ignore
-        return retval
+        if isinstance(obj, type(None)):
+            raise PdfReadError("trailer not found or does not point to document information directory")
+        else:
+            retval.update(obj)  # type: ignore
+            return retval
```

Comment:
```suggestion
        retval.update(obj)  # type: ignore
        return retval
```

Reply:
I'm curious about this one. Why remove the else statement? Is it for performance or something else? Should I have a try statement in place of the if?